app/google_calendar/google_calendar_api.py

```diff
@@ -0,0 +1,243 @@
+from config.settings import settings
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+from models.schemas import CalendarEvent
+
+logger = setup_logging(__name__)
+
+# Path to the Google service account credentials
+_CURRENT_FOLDER = Path(__file__).parent.absolute()
+_DEFAULT_CREDENTIALS_FILE_PATH = _CURRENT_FOLDER / "credentials.json"
+
+SCOPES = ["https://www.googleapis.com/auth/calendar"]
+
+
+@dataclass
+class GoogleCalendarService:
+    """Handles interactions with Google Calendar API for a specific calendar."""
+
+    calendar_id: str
+    service: object | None = field(default=None, init=False)
+
+    def __post_init__(self):
+        """Initialize Google Calendar API service."""
+        try:
+            self.init_credentials_json()
+            creds = service_account.Credentials.from_service_account_file(
+                _DEFAULT_CREDENTIALS_FILE_PATH, scopes=SCOPES
+            )
+            self.service = build("calendar", "v3", credentials=creds)
+            logger.info(f"Authenticated with Google Calendar API for {self.calendar_id}.")
+        except Exception as e:
+            msg = f"Failed to authenticate with Google Calendar API for {self.calendar_id}: {e}"
+            logger.error(msg)
+            self.service = None
+            raise e
+
+    def init_credentials_json(self):
+        """Write to credentials.json from environment variable."""
+        creds = settings.GOOGLE_CREDENTIALS
+        logger.info(f"Writing to credentials.json from memory, json size: {len(creds)}")
+        _DEFAULT_CREDENTIALS_FILE_PATH.write_text(creds)
+
+    def create_event(self, event: CalendarEvent) -> str | bool:
+        """Create a Google Calendar event for a match."""
+
+        try:
+            created_event = (
+                self.service.events()  # type: ignore
+                .insert(calendarId=self.calendar_id, body=event.model_dump())
+                .execute()
+            )
+            logger.info(f"Event created in {self.calendar_id}: {created_event.get('htmlLink')}")
+            return created_event.get("id")
+        except Exception as e:
+            logger.error(f"Failed to create event in {self.calendar_id}: {e}")
+            return False
+
+    def delete_event(self, event_id: str) -> bool:
+        """Delete an event by its ID."""
+        if not self.service:
+            logger.error(
+                f"Google Calendar service not available for {self.calendar_id}. Skipping event deletion."
+            )
+            return False
+
+        try:
+            self.service.events().delete(calendarId=self.calendar_id, eventId=event_id).execute()  # type: ignore
+            logger.info(f"Event {event_id} deleted from {self.calendar_id}.")
+            return True
+        except Exception as e:
+            logger.error(f"Failed to delete event {event_id} from {self.calendar_id}: {e}")
+            return False
+
+    def delete_all_events(self) -> bool:
```

Comment:
Not sure that this method needs to return bool, it should just raise if ther's an error

Reply:
שני המקרים קצת בעייתיים. אפשר לבנות מנגנון יותר מתוחכם אבל לא שווה את הזמן.
שיניתי כך שתיזרק שגיאה

Reply:
למה לזרוק שגיאה זה בעייתי?
נראה שזה עדיין מחזיר בוליאני